# Ultralytics Framework for YOLO11 Training

In [1]:
import torch

# Number of GPUs available
num_gpus = torch.cuda.device_count()

print(f"Number of available GPUs: {num_gpus}")


Number of available GPUs: 4


# Yolo11s Model Training on:
- Carla_Labelling-v5 dataset
- Train: 70% = 4485 Images
- Validation: 20% = 1274 Images
- Test: 10% = 641 Images

- Hyperparameters are on automatic settings using Ultralytics package

In [2]:
from ultralytics import YOLO

# Load YOLOv10n model from scratch
model = YOLO("yolo11s.pt")

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Creating new Ultralytics Settings v0.0.6 file ✅ 
View Ultralytics Settings with 'yolo settings' or at '/home/sur06423/.config/Ultralytics/settings.json'
Update Settings with 'yolo settings key=value', i.e. 'yolo settings runs_dir=path/to/dir'. For help see https://docs.ultralytics.com/quickstart/#ultralytics-settings.


100%|██████████| 18.4M/18.4M [00:00<00:00, 196MB/s]


# Validate the pretrained weights by running inference on an image.

In [3]:
# Run batched inference on a list of images
results = model(["/home/sur06423/project/1.jpg"])  # return a list of Results objects

# Process results list
for result in results:
    boxes = result.boxes  # Boxes object for bounding box outputs
    masks = result.masks  # Masks object for segmentation masks outputs
    keypoints = result.keypoints  # Keypoints object for pose outputs
    probs = result.probs  # Probs object for classification outputs
    obb = result.obb  # Oriented boxes object for OBB outputs
    result.show()  # display to screen
    result.save(filename="result.jpg")  # save to disk


0: 448x640 4 horses, 72.3ms
Speed: 7.3ms preprocess, 72.3ms inference, 489.8ms postprocess per image at shape (1, 3, 448, 640)


Error: no "view" rule for type "image/png" passed its test case
       (for more information, add "--debug=1" on the command line)


# Train the model using the following guidelines:
- Link for various arguments: https://docs.ultralytics.com/modes/train/#train-settings 

In [4]:
import os

# Set library paths
os.environ['LD_LIBRARY_PATH'] = '/usr/lib/xorg-nvidia-525.116.04/lib/x86_64-linux-gnu:' + os.environ.get('LD_LIBRARY_PATH', '')
os.environ['LD_LIBRARY_PATH'] = '/usr/lib/xorg/lib/x86_64-linux-gnu:' + os.environ['LD_LIBRARY_PATH']
os.environ['LD_LIBRARY_PATH'] = '/usr/lib/xorg-nvidia-535.113.01/lib/x86_64-linux-gnu:' + os.environ['LD_LIBRARY_PATH']

# Verify the update
print(os.environ['LD_LIBRARY_PATH'])


/usr/lib/xorg-nvidia-535.113.01/lib/x86_64-linux-gnu:/usr/lib/xorg/lib/x86_64-linux-gnu:/usr/lib/xorg-nvidia-525.116.04/lib/x86_64-linux-gnu:/home/sur06423/miniconda3/envs/yolov_env/lib/python3.9/site-packages/cv2/../../lib64:


In [5]:
# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
model.train(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", # 	Path to the dataset configuration file (e.g., coco8.yaml). This file contains dataset-specific parameters, including paths to training and validation data, class names, and number of classes.
            epochs=100, # Total number of training epochs. Each epoch represents a full pass over the entire dataset. Adjusting this value can affect training duration and model performance.
            patience=10, # Number of epochs to wait without improvement in validation metrics before early stopping the training. Helps prevent overfitting by stopping training when performance plateaus.
            batch=128, # Batch size, with three modes: set as an integer (e.g., batch=16), auto mode for 60% GPU memory utilization (batch=-1), or auto mode with specified utilization fraction (batch=0.70).
            imgsz=640, # Target image size for training. All images are resized to this dimension before being fed into the model. Affects model accuracy and computational complexity.
            save=True, # Enables saving of training checkpoints and final model weights. Useful for resuming training or model deployment.
            save_period = 1, # Frequency of saving model checkpoints, specified in epochs. A value of -1 disables this feature. Useful for saving interim models during long training sessions.
            device=[0,1,2,3], # Specifies the computational device(s) for training: a single GPU (device=0), multiple GPUs (device=0,1), CPU (device=cpu), or MPS for Apple silicon (device=mps).
            workers=6, # Number of worker threads for data loading (per RANK if Multi-GPU training). Influences the speed of data preprocessing and feeding into the model, especially useful in multi-GPU setups.
            project="/home/sur06423/project/Version_5_exp/YOLO_11/v11_s", # Name of the project directory where training outputs are saved. Allows for organized storage of different experiments.
            pretrained = True, # Determines whether to start training from a pretrained model. Can be a boolean value or a string path to a specific model from which to load weights. Enhances training efficiency and model performance.
            optimizer = "auto", # Choice of optimizer for training. Options include SGD, Adam, AdamW, NAdam, RAdam, RMSProp etc., or auto for automatic selection based on model configuration. Affects convergence speed and stability.
            verbose = True, # Enables verbose output during training, providing detailed logs and progress updates. Useful for debugging and closely monitoring the training process.
            seed = 0, # Sets the random seed for training, ensuring reproducibility of results across runs with the same configurations.
            amp = True, # Enables Automatic Mixed Precision (AMP) training, reducing memory usage and possibly speeding up training with minimal impact on accuracy.
            profile =True, # Enables profiling of ONNX and TensorRT speeds during training, useful for optimizing model deployment.
            val=True, # Enables validation during training, allowing for periodic evaluation of model performance on a separate dataset.
            plots = True, #Generates and saves plots of training and validation metrics, as well as prediction examples, providing visual insights into model performance and learning progression.                
)

Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
                                                      CUDA:1 (NVIDIA RTX A6000, 48677MiB)
                                                      CUDA:2 (NVIDIA RTX A6000, 48677MiB)
                                                      CUDA:3 (NVIDIA RTX A6000, 48677MiB)
engine/trainer: task=detect, mode=train, model=yolo11s.pt, data=/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml, epochs=100, time=None, patience=10, batch=128, imgsz=640, save=True, save_period=1, cache=False, device=[0, 1, 2, 3], workers=6, project=/home/sur06423/project/Version_5_exp/YOLO_11/v11_s, name=train, exist_ok=False, pretrained=True, optimizer=auto, verbose=True, seed=0, deterministic=True, single_cls=False, rect=False, cos_lr=False, close_mosaic=10, resume=False, amp=True, fraction=1.0, profile=True, freeze=None, multi_scale=False, overlap_mask=True, mask_ratio=4, dropout=0.0, val=True, split=val, save_jso

Overriding model.yaml nc=80 with nc=4

                   from  n    params  module                                       arguments                     
  0                  -1  1       928  ultralytics.nn.modules.conv.Conv             [3, 32, 3, 2]                 
  1                  -1  1     18560  ultralytics.nn.modules.conv.Conv             [32, 64, 3, 2]                
  2                  -1  1     26080  ultralytics.nn.modules.block.C3k2            [64, 128, 1, False, 0.25]     
  3                  -1  1    147712  ultralytics.nn.modules.conv.Conv             [128, 128, 3, 2]              
  4                  -1  1    103360  ultralytics.nn.modules.block.C3k2            [128, 256, 1, False, 0.25]    
  5                  -1  1    590336  ultralytics.nn.modules.conv.Conv             [256, 256, 3, 2]              
  6                  -1  1    346112  ultralytics.nn.modules.block.C3k2            [256, 256, 1, True]           
  7                  -1  1   1180672  ultralytics

100%|██████████| 5.35M/5.35M [00:00<00:00, 142MB/s]


AMP: checks passed ✅


train: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/train/labels.cache... 4485 images, 430 backgrounds, 0 corrupt: 100%|██████████| 4485/4485 [00:00<?, ?it/s]
val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/valid/labels.cache... 1274 images, 124 backgrounds, 0 corrupt: 100%|██████████| 1274/1274 [00:00<?, ?it/s]


Plotting labels to /home/sur06423/project/Version_5_exp/YOLO_11/v11_s/train/labels.jpg... 
optimizer: 'optimizer=auto' found, ignoring 'lr0=0.01' and 'momentum=0.937' and determining best 'optimizer', 'lr0' and 'momentum' automatically... 
optimizer: AdamW(lr=0.000714, momentum=0.9) with parameter groups 81 weight(decay=0.0), 88 weight(decay=0.001), 87 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 24 dataloader workers
Logging results to /home/sur06423/project/Version_5_exp/YOLO_11/v11_s/train
Starting training for 100 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      1/100      8.34G      1.111      2.495      1.075          5        640: 100%|██████████| 36/36 [00:11<00:00,  3.08it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.58it/s]


                   all       1274       7377      0.722      0.373      0.413      0.266

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      2/100      8.45G     0.9299     0.7919     0.9693         17        640: 100%|██████████| 36/36 [00:09<00:00,  3.71it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.05it/s]


                   all       1274       7377      0.737      0.428      0.473      0.278

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      3/100      8.42G     0.9553     0.7634     0.9952         37        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.02it/s]


                   all       1274       7377      0.578      0.414      0.385      0.221

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      4/100      8.37G     0.9503     0.7273      1.007          8        640: 100%|██████████| 36/36 [00:09<00:00,  3.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.10it/s]


                   all       1274       7377      0.551      0.455      0.432       0.24

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      5/100      8.37G     0.9191     0.7031     0.9744          6        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.99it/s]


                   all       1274       7377      0.677      0.421      0.443      0.258

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      6/100      8.37G     0.9024     0.6577     0.9656         26        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.08it/s]


                   all       1274       7377      0.679      0.472      0.494      0.289

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      7/100      8.45G     0.8905     0.6282     0.9607          3        640: 100%|██████████| 36/36 [00:08<00:00,  4.01it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.12it/s]


                   all       1274       7377      0.695      0.485      0.509      0.307

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      8/100      8.38G     0.8736     0.6194     0.9413         24        640: 100%|██████████| 36/36 [00:09<00:00,  3.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.15it/s]


                   all       1274       7377      0.779      0.498      0.557      0.338

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      9/100       8.4G     0.8292     0.5852     0.9434          1        640: 100%|██████████| 36/36 [00:09<00:00,  3.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.95it/s]


                   all       1274       7377      0.693      0.517      0.541       0.33

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     10/100      8.35G     0.8517     0.5829      0.938          8        640: 100%|██████████| 36/36 [00:09<00:00,  3.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.05it/s]


                   all       1274       7377      0.796      0.501      0.568      0.351

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     11/100      8.36G     0.8655     0.5706     0.9294         14        640: 100%|██████████| 36/36 [00:09<00:00,  3.87it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.05it/s]


                   all       1274       7377       0.81      0.521      0.586      0.355

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     12/100      8.38G     0.8534     0.6588     0.9215          1        640: 100%|██████████| 36/36 [00:09<00:00,  3.87it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.95it/s]


                   all       1274       7377      0.804      0.508      0.577      0.357

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     13/100      8.44G     0.8279     0.5529     0.9271         11        640: 100%|██████████| 36/36 [00:09<00:00,  3.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.02it/s]


                   all       1274       7377      0.821      0.534      0.604      0.367

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     14/100      8.43G     0.8268     0.5634     0.9182         14        640: 100%|██████████| 36/36 [00:09<00:00,  3.90it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.15it/s]


                   all       1274       7377      0.846      0.539      0.618      0.385

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     15/100      8.38G      0.817     0.5338     0.9189         24        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.12it/s]


                   all       1274       7377      0.825      0.548      0.609      0.378

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     16/100      8.42G     0.8056     0.5269     0.9194          4        640: 100%|██████████| 36/36 [00:09<00:00,  3.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.09it/s]


                   all       1274       7377      0.839      0.546      0.626        0.4

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     17/100      8.43G      0.817      0.544     0.9293         15        640: 100%|██████████| 36/36 [00:09<00:00,  3.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.10it/s]


                   all       1274       7377      0.836      0.544      0.628      0.393

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     18/100      8.37G     0.7887     0.5225     0.9311          6        640: 100%|██████████| 36/36 [00:09<00:00,  3.90it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.09it/s]


                   all       1274       7377      0.837      0.551      0.624      0.384

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     19/100      8.42G     0.7995     0.5119     0.9158         12        640: 100%|██████████| 36/36 [00:09<00:00,  3.79it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.08it/s]


                   all       1274       7377      0.861      0.536      0.633       0.39

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     20/100      8.41G     0.7796      0.505     0.9147         23        640: 100%|██████████| 36/36 [00:09<00:00,  3.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.10it/s]


                   all       1274       7377       0.81      0.564      0.619      0.391

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     21/100      8.43G     0.7774      0.512     0.9122          4        640: 100%|██████████| 36/36 [00:09<00:00,  3.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.16it/s]


                   all       1274       7377      0.824      0.567      0.636      0.401

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     22/100      8.41G     0.8019     0.5259     0.9106          3        640: 100%|██████████| 36/36 [00:09<00:00,  3.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.08it/s]


                   all       1274       7377      0.832      0.571      0.634      0.393

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     23/100      8.47G     0.7594     0.4891     0.9097         27        640: 100%|██████████| 36/36 [00:08<00:00,  4.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.15it/s]


                   all       1274       7377      0.829      0.582      0.647      0.414

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     24/100      8.38G     0.7552     0.4765     0.9052          3        640: 100%|██████████| 36/36 [00:08<00:00,  4.04it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.08it/s]


                   all       1274       7377      0.835       0.57      0.641      0.397

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     25/100      8.39G     0.7586     0.4782     0.9078         20        640: 100%|██████████| 36/36 [00:09<00:00,  3.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.11it/s]


                   all       1274       7377      0.848      0.584       0.66      0.415

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     26/100      8.44G     0.7437     0.4709     0.9124         18        640: 100%|██████████| 36/36 [00:09<00:00,  3.90it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.09it/s]


                   all       1274       7377      0.825      0.584      0.646      0.407

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     27/100      8.42G     0.7721     0.4728     0.9045         11        640: 100%|██████████| 36/36 [00:09<00:00,  3.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.04it/s]


                   all       1274       7377      0.852       0.57       0.65      0.411

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     28/100      8.35G     0.7394     0.4699     0.9006         20        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.12it/s]


                   all       1274       7377      0.815      0.598      0.657      0.414

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     29/100      8.41G     0.7265     0.4509     0.8934          7        640: 100%|██████████| 36/36 [00:09<00:00,  3.96it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.17it/s]


                   all       1274       7377      0.825      0.598      0.668      0.424

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     30/100      8.37G     0.7467     0.4582     0.9052         11        640: 100%|██████████| 36/36 [00:09<00:00,  3.90it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.15it/s]


                   all       1274       7377      0.862      0.584      0.664      0.421

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     31/100      8.39G     0.7195     0.4483     0.8942         19        640: 100%|██████████| 36/36 [00:09<00:00,  3.97it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.10it/s]


                   all       1274       7377       0.85      0.593      0.666      0.422

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     32/100      8.41G     0.7335     0.4665      0.893         10        640: 100%|██████████| 36/36 [00:09<00:00,  3.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.73it/s]


                   all       1274       7377      0.832      0.598      0.661      0.415

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     33/100      8.38G     0.7141     0.4446     0.8922         13        640: 100%|██████████| 36/36 [00:08<00:00,  4.02it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.18it/s]


                   all       1274       7377      0.875      0.587      0.676      0.432

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     34/100      8.42G     0.7215     0.4411     0.8971         11        640: 100%|██████████| 36/36 [00:09<00:00,  3.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.02it/s]


                   all       1274       7377       0.84      0.573      0.657      0.414

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     35/100      8.41G     0.7357     0.4478     0.8989         21        640: 100%|██████████| 36/36 [00:09<00:00,  3.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.16it/s]


                   all       1274       7377       0.84      0.608      0.674      0.432

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     36/100      8.42G     0.7142     0.4418     0.8922          9        640: 100%|██████████| 36/36 [00:08<00:00,  4.00it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.09it/s]


                   all       1274       7377      0.846      0.603      0.678      0.434

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     37/100      8.42G     0.7182     0.4441     0.8952         10        640: 100%|██████████| 36/36 [00:09<00:00,  3.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.10it/s]


                   all       1274       7377      0.864      0.611      0.682      0.437

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     38/100      8.42G     0.6881     0.6986     0.8753          0        640: 100%|██████████| 36/36 [00:09<00:00,  3.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.13it/s]


                   all       1274       7377      0.866      0.607      0.682      0.436

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     39/100      8.39G     0.6966     0.4272     0.8894         21        640: 100%|██████████| 36/36 [00:09<00:00,  3.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.18it/s]


                   all       1274       7377      0.854      0.604      0.682      0.434

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     40/100      8.36G     0.7045     0.4337     0.8868         15        640: 100%|██████████| 36/36 [00:09<00:00,  3.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.06it/s]


                   all       1274       7377      0.864      0.609      0.685      0.446

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     41/100      8.38G     0.6901     0.4317     0.8894         10        640: 100%|██████████| 36/36 [00:08<00:00,  4.04it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.15it/s]


                   all       1274       7377      0.872      0.609      0.693      0.439

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     42/100      8.41G      0.693     0.4296     0.8889         24        640: 100%|██████████| 36/36 [00:09<00:00,  3.93it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.12it/s]


                   all       1274       7377      0.869      0.615      0.692      0.447

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     43/100      8.38G     0.6774     0.4257     0.8926          4        640: 100%|██████████| 36/36 [00:09<00:00,  3.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.10it/s]


                   all       1274       7377      0.867      0.611      0.692      0.447

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     44/100      8.67G     0.7057     0.4201     0.8768         26        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.07it/s]


                   all       1274       7377      0.864        0.6      0.685      0.437

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     45/100      8.38G     0.6728      0.411     0.8814          8        640: 100%|██████████| 36/36 [00:08<00:00,  4.01it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.16it/s]


                   all       1274       7377       0.84      0.629      0.695      0.446

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     46/100      8.41G     0.6641     0.4087     0.8805          8        640: 100%|██████████| 36/36 [00:09<00:00,  3.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.04it/s]


                   all       1274       7377      0.863      0.617      0.693      0.448

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     47/100      8.37G      0.669     0.4023     0.8778         15        640: 100%|██████████| 36/36 [00:09<00:00,  3.92it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  4.00it/s]


                   all       1274       7377      0.876      0.608      0.697      0.448

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     48/100      8.42G      0.683     0.4022      0.879         16        640: 100%|██████████| 36/36 [00:09<00:00,  3.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.20it/s]


                   all       1274       7377      0.855      0.624      0.694      0.449

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     49/100      8.37G     0.7025     0.4333     0.8632          7        640: 100%|██████████| 36/36 [00:09<00:00,  3.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.13it/s]


                   all       1274       7377      0.861      0.628      0.703      0.455

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     50/100      8.36G     0.6855     0.4078     0.8757         17        640: 100%|██████████| 36/36 [00:09<00:00,  3.96it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.95it/s]


                   all       1274       7377      0.874      0.612        0.7      0.455

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     51/100      8.43G     0.6743     0.4029     0.8884         10        640: 100%|██████████| 36/36 [00:09<00:00,  3.92it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.09it/s]


                   all       1274       7377      0.851      0.632      0.707      0.461

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     52/100      8.41G     0.6626     0.3928     0.8835          5        640: 100%|██████████| 36/36 [00:09<00:00,  3.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.98it/s]


                   all       1274       7377       0.86      0.629      0.702      0.453

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     53/100      8.37G     0.6554      0.394      0.881          8        640: 100%|██████████| 36/36 [00:08<00:00,  4.02it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.12it/s]


                   all       1274       7377      0.878      0.612      0.705      0.459

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     54/100      8.34G     0.6368      0.385     0.8729         14        640: 100%|██████████| 36/36 [00:09<00:00,  3.96it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.18it/s]


                   all       1274       7377       0.84      0.632      0.703       0.46

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     55/100      8.42G     0.6572     0.3923     0.8752         21        640: 100%|██████████| 36/36 [00:09<00:00,  3.86it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.10it/s]


                   all       1274       7377      0.882      0.603        0.7      0.454

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     56/100      8.39G      0.643     0.3812     0.8771          5        640: 100%|██████████| 36/36 [00:08<00:00,  4.02it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.11it/s]


                   all       1274       7377      0.867      0.623      0.703      0.455

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     57/100      8.44G     0.6561     0.3861      0.868         13        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.11it/s]


                   all       1274       7377      0.869      0.628      0.713      0.466

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     58/100      8.39G     0.6443      0.383      0.876         21        640: 100%|██████████| 36/36 [00:09<00:00,  3.92it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.09it/s]


                   all       1274       7377      0.864       0.63      0.715       0.47

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     59/100      8.36G     0.6508     0.3823     0.8724         12        640: 100%|██████████| 36/36 [00:09<00:00,  3.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.15it/s]


                   all       1274       7377      0.856      0.643      0.718      0.469

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     60/100      8.42G     0.6489     0.3827     0.8743         18        640: 100%|██████████| 36/36 [00:09<00:00,  3.86it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.14it/s]


                   all       1274       7377      0.878      0.627      0.717      0.473

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     61/100      8.37G     0.6512      0.382     0.8705         39        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.10it/s]


                   all       1274       7377      0.853      0.636      0.712      0.463

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     62/100      8.37G     0.6289     0.3737     0.8711         33        640: 100%|██████████| 36/36 [00:09<00:00,  3.96it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.11it/s]


                   all       1274       7377      0.881      0.622      0.711      0.466

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     63/100      8.34G      0.647     0.3836     0.8705         13        640: 100%|██████████| 36/36 [00:09<00:00,  3.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.13it/s]


                   all       1274       7377      0.879      0.624      0.715      0.472

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     64/100      8.39G     0.6637     0.3828     0.8678          6        640: 100%|██████████| 36/36 [00:08<00:00,  4.00it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.94it/s]


                   all       1274       7377       0.87      0.632      0.722      0.476

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     65/100      8.41G     0.6603     0.3885      0.871          3        640: 100%|██████████| 36/36 [00:09<00:00,  3.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.15it/s]


                   all       1274       7377      0.882      0.626      0.721      0.477

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     66/100      8.37G     0.6404     0.3734     0.8746         10        640: 100%|██████████| 36/36 [00:09<00:00,  3.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.13it/s]


                   all       1274       7377       0.86      0.639      0.722      0.474

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     67/100      8.42G     0.6442     0.3756     0.8702         19        640: 100%|██████████| 36/36 [00:09<00:00,  3.92it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.10it/s]


                   all       1274       7377      0.851      0.638      0.719      0.474

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     68/100      8.39G     0.6204     0.3656     0.8605         19        640: 100%|██████████| 36/36 [00:09<00:00,  3.87it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.50it/s]


                   all       1274       7377      0.878      0.626      0.715      0.477

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     69/100      8.43G     0.6432     0.3752     0.8719         16        640: 100%|██████████| 36/36 [00:09<00:00,  3.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.10it/s]


                   all       1274       7377      0.848      0.649      0.716      0.474

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     70/100      8.42G     0.6306     0.3675     0.8696         24        640: 100%|██████████| 36/36 [00:09<00:00,  3.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.08it/s]


                   all       1274       7377      0.853       0.64      0.713      0.473

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     71/100      8.42G     0.6258     0.3559     0.8676         17        640: 100%|██████████| 36/36 [00:09<00:00,  3.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.06it/s]


                   all       1274       7377      0.889       0.63      0.722      0.479

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     72/100      8.38G     0.6239     0.3622     0.8656         12        640: 100%|██████████| 36/36 [00:09<00:00,  3.89it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.06it/s]


                   all       1274       7377      0.869      0.638      0.721      0.481

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     73/100      8.37G      0.619     0.3563     0.8681         13        640: 100%|██████████| 36/36 [00:09<00:00,  3.88it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.14it/s]


                   all       1274       7377      0.858      0.639      0.724      0.481

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     74/100      8.43G     0.6135     0.3536     0.8714         11        640: 100%|██████████| 36/36 [00:09<00:00,  3.89it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.13it/s]


                   all       1274       7377      0.857      0.646      0.718      0.478

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     75/100      8.42G     0.6228     0.3589      0.864          6        640: 100%|██████████| 36/36 [00:09<00:00,  3.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.67it/s]


                   all       1274       7377       0.87      0.641      0.723      0.481

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     76/100      8.41G     0.6263     0.3537      0.858          8        640: 100%|██████████| 36/36 [00:09<00:00,  3.90it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.13it/s]


                   all       1274       7377       0.88      0.641      0.725      0.482

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     77/100      8.38G     0.6195     0.3552      0.861         22        640: 100%|██████████| 36/36 [00:09<00:00,  3.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.16it/s]


                   all       1274       7377      0.847      0.654      0.723       0.48

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     78/100       8.4G     0.6161     0.3488     0.8733          4        640: 100%|██████████| 36/36 [00:09<00:00,  3.91it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.08it/s]


                   all       1274       7377      0.861      0.653      0.728      0.484

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     79/100      8.36G     0.6154     0.3492     0.8669         13        640: 100%|██████████| 36/36 [00:09<00:00,  3.94it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.11it/s]


                   all       1274       7377      0.857      0.656      0.728      0.483

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     80/100      8.42G     0.6024     0.3488      0.861         36        640: 100%|██████████| 36/36 [00:09<00:00,  3.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.16it/s]


                   all       1274       7377      0.863      0.657      0.733      0.486

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     81/100       8.4G     0.6058     0.3469      0.865          7        640: 100%|██████████| 36/36 [00:09<00:00,  3.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.20it/s]


                   all       1274       7377      0.893      0.637      0.727      0.485

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     82/100      8.43G     0.5989     0.3453     0.8588          9        640: 100%|██████████| 36/36 [00:09<00:00,  3.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.18it/s]


                   all       1274       7377      0.877      0.644      0.732      0.489

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     83/100      8.42G     0.6194     0.3437     0.8609          8        640: 100%|██████████| 36/36 [00:09<00:00,  3.82it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.13it/s]


                   all       1274       7377      0.884       0.65      0.737      0.492

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     84/100      8.37G     0.6101     0.3534     0.8615          8        640: 100%|██████████| 36/36 [00:09<00:00,  3.90it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.12it/s]


                   all       1274       7377       0.88      0.649      0.735      0.493

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     85/100       8.4G     0.6112     0.3476     0.8697         11        640: 100%|██████████| 36/36 [00:09<00:00,  3.78it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.93it/s]


                   all       1274       7377      0.888      0.641      0.733       0.49

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     86/100      8.38G     0.6322     0.4415     0.8932          3        640: 100%|██████████| 36/36 [00:09<00:00,  3.87it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.17it/s]


                   all       1274       7377      0.879      0.644      0.732       0.49

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     87/100      8.37G     0.6106     0.3377     0.8658         14        640: 100%|██████████| 36/36 [00:09<00:00,  3.87it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.13it/s]


                   all       1274       7377       0.88      0.651      0.736       0.49

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     88/100      8.41G     0.6001     0.3383     0.8601         16        640: 100%|██████████| 36/36 [00:09<00:00,  3.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.13it/s]


                   all       1274       7377      0.868      0.656      0.737      0.495

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     89/100      8.36G      0.569     0.3198      0.862         14        640: 100%|██████████| 36/36 [00:09<00:00,  3.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.05it/s]


                   all       1274       7377      0.862      0.659      0.736      0.495

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     90/100      8.39G     0.5809     0.3339      0.867          3        640: 100%|██████████| 36/36 [00:09<00:00,  3.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.20it/s]


                   all       1274       7377       0.89      0.643      0.738      0.496
Closing dataloader mosaic

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     91/100      8.41G     0.6509     0.3418     0.8472         13        640: 100%|██████████| 36/36 [00:10<00:00,  3.56it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:05<00:00,  3.74it/s]


                   all       1274       7377      0.877       0.65      0.734      0.491

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     92/100      8.39G     0.6436     0.3377     0.8475         13        640: 100%|██████████| 36/36 [00:08<00:00,  4.01it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.06it/s]


                   all       1274       7377      0.894      0.646      0.736      0.493

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     93/100      8.38G     0.6434     0.3428     0.8438          7        640: 100%|██████████| 36/36 [00:09<00:00,  3.90it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.15it/s]


                   all       1274       7377      0.875      0.658      0.739      0.498

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     94/100      8.42G     0.6524     0.3449     0.8475          6        640: 100%|██████████| 36/36 [00:09<00:00,  3.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.15it/s]


                   all       1274       7377      0.887      0.651      0.737      0.497

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     95/100      8.45G     0.6562      0.345     0.8433         31        640: 100%|██████████| 36/36 [00:09<00:00,  3.98it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.13it/s]


                   all       1274       7377      0.868      0.659      0.736      0.495

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     96/100      8.42G     0.6432     0.3367     0.8481         13        640: 100%|██████████| 36/36 [00:09<00:00,  3.95it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.16it/s]


                   all       1274       7377      0.883      0.656       0.74      0.497

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     97/100      8.42G     0.6473     0.3374     0.8447         13        640: 100%|██████████| 36/36 [00:09<00:00,  3.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.18it/s]


                   all       1274       7377      0.878      0.653       0.74      0.498

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     98/100      8.42G     0.6396     0.3402     0.8371          3        640: 100%|██████████| 36/36 [00:08<00:00,  4.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.18it/s]


                   all       1274       7377       0.87      0.657       0.74      0.499

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     99/100      8.37G     0.6362     0.3331     0.8425         40        640: 100%|██████████| 36/36 [00:08<00:00,  4.07it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.17it/s]


                   all       1274       7377      0.868      0.658      0.739      0.499

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    100/100      8.39G       0.63     0.3321     0.8432          6        640: 100%|██████████| 36/36 [00:08<00:00,  4.02it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:04<00:00,  4.16it/s]


                   all       1274       7377      0.889      0.646      0.737      0.499

100 epochs completed in 0.503 hours.
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_11/v11_s/train/weights/last.pt, 19.2MB
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_11/v11_s/train/weights/best.pt, 19.2MB

Validating /home/sur06423/project/Version_5_exp/YOLO_11/v11_s/train/weights/best.pt...
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
                                                      CUDA:1 (NVIDIA RTX A6000, 48677MiB)
                                                      CUDA:2 (NVIDIA RTX A6000, 48677MiB)
                                                      CUDA:3 (NVIDIA RTX A6000, 48677MiB)
YOLO11s summary (fused): 238 layers, 9,414,348 parameters, 0 gradients, 21.3 GFLOPs


                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:07<00:00,  2.60it/s]


                   all       1274       7377      0.884       0.65      0.737      0.499
            Pedestrian        333        600      0.811      0.675      0.751      0.452
         Traffic_Signs        309        434      0.896       0.59      0.676      0.423
               Vehicle       1034       2820      0.929       0.88      0.931      0.755
         traffic_light        660       3523        0.9      0.457      0.588      0.367
Speed: 0.3ms preprocess, 1.5ms inference, 0.0ms loss, 0.7ms postprocess per image
Results saved to /home/sur06423/project/Version_5_exp/YOLO_11/v11_s/train


# Results on Validation set:
100 epochs completed in 0.503 hours.
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_11/v11_s/train/weights/last.pt, 19.2MB
Optimizer stripped from /home/sur06423/project/Version_5_exp/YOLO_11/v11_s/train/weights/best.pt, 19.2MB

Validating /home/sur06423/project/Version_5_exp/YOLO_11/v11_s/train/weights/best.pt...
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
                                                      CUDA:1 (NVIDIA RTX A6000, 48677MiB)
                                                      CUDA:2 (NVIDIA RTX A6000, 48677MiB)
                                                      CUDA:3 (NVIDIA RTX A6000, 48677MiB)
YOLO11s summary (fused): 238 layers, 9,414,348 parameters, 0 gradients, 21.3 GFLOPs
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 20/20 [00:07<00:00,  2.60it/s]
                   all       1274       7377      0.884       0.65      0.737      0.499
            Pedestrian        333        600      0.811      0.675      0.751      0.452
         Traffic_Signs        309        434      0.896       0.59      0.676      0.423
               Vehicle       1034       2820      0.929       0.88      0.931      0.755
         traffic_light        660       3523        0.9      0.457      0.588      0.367
Speed: 0.3ms preprocess, 1.5ms inference, 0.0ms loss, 0.7ms postprocess per image
Results saved to /home/sur06423/project/Version_5_exp/YOLO_11/v11_s/train

# Test the results on Test dataset
- The above results was on the validation set of the Carla_labeling-v5 dataset (See the Image counts)
- Now we need to calculate the metrics on the test set of the Carla_labeling-v5 dataset containing only 641 images.
- Below is the script to do so:

In [6]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load YOLOv10s Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_11/v11_s/train/weights/last.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.25, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLO11s summary (fused): 238 layers, 9,414,348 parameters, 0 gradients, 21.3 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:10<00:00,  1.68s/it]


                   all        641       3701      0.873      0.643      0.787      0.575
            Pedestrian        182        332       0.77      0.666      0.775      0.512
         Traffic_Signs        168        230      0.905      0.583      0.765      0.523
               Vehicle        528       1377      0.938      0.877      0.931      0.804
         traffic_light        338       1762       0.88      0.449      0.678      0.461
Speed: 0.3ms preprocess, 2.6ms inference, 0.0ms loss, 2.0ms postprocess per image
Saving runs/detect/val/predictions.json...
Results saved to runs/detect/val


# Results on Last checpoint (100th epoch):
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLO11s summary (fused): 238 layers, 9,414,348 parameters, 0 gradients, 21.3 GFLOPs
val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:10<00:00,  1.68s/it]
                   all        641       3701      0.873      0.643      0.787      0.575
            Pedestrian        182        332       0.77      0.666      0.775      0.512
         Traffic_Signs        168        230      0.905      0.583      0.765      0.523
               Vehicle        528       1377      0.938      0.877      0.931      0.804
         traffic_light        338       1762       0.88      0.449      0.678      0.461
Speed: 0.3ms preprocess, 2.6ms inference, 0.0ms loss, 2.0ms postprocess per image
Saving runs/detect/val/predictions.json...
Results saved to runs/detect/val

# Results on Test Set using Best Checkpoint:

In [7]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load the model
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_11/v11_s/train/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.25, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLO11s summary (fused): 238 layers, 9,414,348 parameters, 0 gradients, 21.3 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:09<00:00,  1.52s/it]


                   all        641       3701      0.873      0.643      0.787      0.575
            Pedestrian        182        332       0.77      0.666      0.775      0.512
         Traffic_Signs        168        230      0.905      0.583      0.765      0.523
               Vehicle        528       1377      0.938      0.877      0.931      0.804
         traffic_light        338       1762       0.88      0.449      0.678      0.461
Speed: 0.3ms preprocess, 2.1ms inference, 0.0ms loss, 3.0ms postprocess per image
Saving runs/detect/val2/predictions.json...
Results saved to runs/detect/val2


In [15]:
print(checkpoint.keys())

dict_keys(['date', 'version', 'license', 'docs', 'epoch', 'best_fitness', 'model', 'ema', 'updates', 'optimizer', 'train_args', 'train_metrics', 'train_results'])


# Taking the Optimal value from the F1-Confidence curve for Optimal Confidence
- Optimal Confidence YOLO11s - 0.304
- IoU threshold : 0.5
- Results save d at: runs/detect/val3

In [1]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load YOLOv10s Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_11/v11_s/train/weights/last.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.304, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.5, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLO11s summary (fused): 238 layers, 9,414,348 parameters, 0 gradients, 21.3 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:08<00:00,  1.40s/it]


                   all        641       3701      0.901      0.626      0.785      0.576
            Pedestrian        182        332      0.826      0.645      0.776      0.518
         Traffic_Signs        168        230      0.934      0.557      0.759       0.52
               Vehicle        528       1377       0.95      0.871       0.93      0.804
         traffic_light        338       1762      0.894      0.434      0.675      0.464
Speed: 2.0ms preprocess, 2.5ms inference, 0.0ms loss, 2.4ms postprocess per image
Saving runs/detect/val3/predictions.json...
Results saved to runs/detect/val3


# Best Checkpoint + Taking the Optimal value from the F1-Confidence curve for Optimal Confidence
- Optimal Confidence YOLO11s - 0.304
- IoU threshold : 0.5
- Results save d at: runs/detect/val4

In [1]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load YOLOv10s Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_11/v11_s/train/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.304, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.5, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLO11s summary (fused): 238 layers, 9,414,348 parameters, 0 gradients, 21.3 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:06<00:00,  1.13s/it]


                   all        641       3701      0.901      0.626      0.785      0.576
            Pedestrian        182        332      0.826      0.645      0.776      0.518
         Traffic_Signs        168        230      0.934      0.557      0.759       0.52
               Vehicle        528       1377       0.95      0.871       0.93      0.804
         traffic_light        338       1762      0.894      0.434      0.675      0.464
Speed: 1.9ms preprocess, 2.5ms inference, 0.0ms loss, 1.5ms postprocess per image
Saving runs/detect/val4/predictions.json...
Results saved to runs/detect/val4


# Best Checkpoint + Optimal Confidence + IoU 0.6
- Optimal Confidence YOLO11s - 0.304
- IoU threshold : 0.6
- Results save d at: runs/detect/val5
- Best Epoch - 100

In [1]:
from ultralytics import YOLO

# Load YOLOv10s model from scratch
# model = YOLO("yolov10s.pt")
# Load YOLOv10s Best weights on carla dataset
model = YOLO("/home/sur06423/project/Version_5_exp/YOLO_11/v11_s/train/weights/best.pt")

# Train the model: https://docs.ultralytics.com/modes/train/#train-settings
validation_results = model.val(data="/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml", 
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.304, # Sets the minimum confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="test",              
)

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLO11s summary (fused): 238 layers, 9,414,348 parameters, 0 gradients, 21.3 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/test/labels.cache... 641 images, 53 backgrounds, 0 corrupt: 100%|██████████| 641/641 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 6/6 [00:08<00:00,  1.34s/it]


                   all        641       3701      0.899      0.627      0.785      0.576
            Pedestrian        182        332       0.82      0.645      0.775      0.517
         Traffic_Signs        168        230      0.934      0.557      0.759       0.52
               Vehicle        528       1377      0.947      0.873       0.93      0.804
         traffic_light        338       1762      0.894      0.434      0.675      0.464
Speed: 1.9ms preprocess, 2.5ms inference, 0.0ms loss, 2.0ms postprocess per image
Saving runs/detect/val5/predictions.json...
Results saved to runs/detect/val5


# Seperate evaluation on the training set

In [1]:
from ultralytics import YOLO

# Load your model
model = YOLO('/home/sur06423/project/Version_5_exp/YOLO_11/v11_s/train/weights/best.pt')  # replace with your model path

# Validate on custom dataset
results = model.val(data='/home/sur06423/project/Version_5_exp/Carla_Labeling-5/data.yaml',
            batch=128, 
            imgsz=640,
            save_json = True, # If True, saves the results to a JSON file for further analysis or integration with other tools.
            conf=0.25, # Sets the optimal confidence threshold for detections. Detections with confidence below this threshold are discarded.
            iou=0.6, # Sets the Intersection Over Union (IoU) threshold for Non-Maximum Suppression (NMS). Helps in reducing duplicate detections.
            device=0, 
            split="train", 
)  # Ensure your YAML points to your desired dataset

WARNING ⚠️ Python>=3.10 is required, but Python==3.9.19 is currently installed 
Ultralytics 8.3.13 🚀 Python-3.9.19 torch-2.4.1+cu121 CUDA:0 (NVIDIA RTX A6000, 48677MiB)
YOLO11s summary (fused): 238 layers, 9,414,348 parameters, 0 gradients, 21.3 GFLOPs


val: Scanning /home/sur06423/project/Version_5_exp/Carla_Labeling-5/train/labels.cache... 4485 images, 430 backgrounds, 0 corrupt: 100%|██████████| 4485/4485 [00:00<?, ?it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 36/36 [00:24<00:00,  1.45it/s]


                   all       4485      26788      0.903      0.687      0.822      0.627
            Pedestrian       1229       2227      0.825      0.761      0.848      0.607
         Traffic_Signs       1093       1459      0.926      0.636      0.799      0.557
               Vehicle       3636      10031      0.949      0.905      0.948      0.838
         traffic_light       2373      13071       0.91      0.447      0.692      0.505
Speed: 0.4ms preprocess, 2.1ms inference, 0.0ms loss, 0.6ms postprocess per image
Saving runs/detect/val6/predictions.json...
Results saved to runs/detect/val6


# This notebook is complete.
- All the cell are executed and saved for later use.
- Results are saved in the corresponding directories.